### PR TITLE
feat: Make federation and domain nullable

### PR DESCRIPTION
**Pull request by KaterinaWire**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Clients should be compatible to older backends that don't provide federation and domain.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by agisilaos — APPROVED** [review]

File: Source/SessionManager/APIVersionResolver.swift [hunk 1]
```diff
@@ -103,8 +103,8 @@ final class APIVersionResolver {
     private struct APIVersionResponsePayload: Decodable {
 
         let supported: [Int32]
-        let federation: Bool
-        let domain: String
+        let federation: Bool?
+        let domain: String?
```
Comment (johnxnguyen) [inline Source/SessionManager/APIVersionResolver.swift:107]: Also the domain? For me a nil domain is an error. We could also give a default to wire.com or something like that.

Comment (johnxnguyen) [reply Source/SessionManager/APIVersionResolver.swift:107]: Actually scrap that, maybe it's fine.

**Review by johnxnguyen — APPROVED** [review]

File: Source/SessionManager/APIVersionResolver.swift (hunk 1 above)
Comment (KaterinaWire) [reply Source/SessionManager/APIVersionResolver.swift:107]: I don't know, but domain in Data model is
```
/// The domain of the backend to which the app is connected to.
    public static var domain: String?
```
that's why I decided to make it optional too

**Comment by KaterinaWire** [comment]
release patch

**Comment by zenkins** [comment]
Hello, @KaterinaWire! Merging PR for command **release patch** for PR #**1741**. Will create a **patch** release after merged.
Next version is 426.1.1.

**Pull request merged**